src/backend/distributed/connection/placement_connection.c

```diff
@@ -846,6 +846,18 @@ ConnectionModifiedPlacement(MultiConnection *connection)
 		return false;
 	}
 
+	if (dlist_is_empty(&connection->referencedPlacements))
```

Comment:
This practically brings back the behavior as if before https://github.com/citusdata/citus/pull/4587/files for non-placement connections.

So, should be pretty safe.

Would you mind adding regression tests to https://github.com/citusdata/citus/blob/33c620f232e8c1cc96adbf43f8a04dfd20714013/src/test/regress/sql/multi_transaction_recovery.sql where we rely on a node connection and make sure that it is using 2PC?

Reply:
added